### PR TITLE
Tech debt: New Relic filtering: move up check for JS error filtering

### PR DESCRIPTION
**Pull request by caleybrock**
I moved the New Relic filtering outside of code-studio and into a separate entry point that happens earlier. I was hoping to add it earlier, but got some errors at each level earlier so I figured this was at least a good start.

### CODE REVIEW

File: dashboard/app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -14,6 +14,10 @@
     = javascript_include_tag "js/#{js_locale}/common_locale"
     %script{src: minifiable_asset_path('js/essential.js')}
     %script{src: minifiable_asset_path('js/code-studio-common.js')}
+
+    //Included early to prevent JS Errors from being sent to New Relic
+    %script{src: minifiable_asset_path('js/layouts/application.js')}
+
```
Comment (Bjvanminnen) [inline dashboard/app/views/layouts/application.html.haml]: Any reason this shouldnt also be before essential/code-studio-common?

Comment (Bjvanminnen) [reply dashboard/app/views/layouts/application.html.haml]: Or alternatively, could we even just make it so that this ends up IN essential I wonder

Comment (joshlory) [reply dashboard/app/views/layouts/application.html.haml]: Some errors (e.g. `UncaughtException: duplicate formal argument e` in very old versions of Firefox) appear to be coming from files we've imported but haven't run yet.  Maybe there's a better workaround?

Comment (joshlory) [reply dashboard/app/views/layouts/application.html.haml]: Here's an example of IE8 throwing when it tries to parse `application.js`:

image

**Comment by caleybrock** [comment]
@Bjvanminnen and @joshlory This now adds the new relic check to the build essential.js by creating an essential.js entry point that will be contanated with what gets build here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/Gruntfile.js#L510

**Comment by joshlory** [comment]
It looks like `essential.js` is included after `application.js`? (See: https://github.com/code-dot-org/code-dot-org/pull/14759#discussion_r114171997.)

**Comment by caleybrock** [comment]
The unsupportedBrowser check won't work before essential.js because my upstanding is essential loads what makes the rest of our JS work.
I'm not sure what the work around is for the Firefox bug, but I think the work around for IE 8 is to add back in the ruby file that filters out old IE versions. I think this work still helps, but it's challenging to catch everything.

**Comment by joshlory** [comment]
Ah, I see — `browser-detector.js` doesn't import anything, but because of the webpack module stuff it won't work without `webpackJsonp` etc?

**Comment by joshlory** [comment]
I'd be tempted to put `essential.js` before `application.js` unless there's an order dependency there.

**Comment by caleybrock** [comment]
@joshlory `essential.js` before `application.js` seems to work, although there is a bit of a concern that at some point something will get added to `essential` that requires `jquery` (right now `application.js` loads `jquery`). This should be obvious that it is broken, but maybe not obvious why.
I made a comment about the order in the haml file to help people if they run into this in the future. Hopefully, we'll start moving things out of `application.js` and into our apps pipeline and this won't be an issue. 
After chatting with @Bjvanminnen we think this is worth it to make this more to reduce noise.

**Pull request merged**